src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -86,6 +88,18 @@ protected function instantiateController($class)
             return $this->container->get($class);
         }
 
-        return parent::instantiateController($class);
+        try {
+            return parent::instantiateController($class);
+        } catch (FatalThrowableError $e) {
+        } catch (\ArgumentCountError $e) {
+        }
+
+        if (null !== $e) {
+            if ($this->container instanceof Container) {
+                if (in_array($class, $this->container->getRemovedIds())) {
```

Comment:
A bit unrelated but if maybe an opportunity to catch character case-mismatches also here? Supposes that we can prevent the parent from failing hard in that case.